### PR TITLE
Issue #838

### PR DESCRIPTION
**Pull request by Fojar**
The spray tool will now skip painting sometimes if it needs to, so overall it achieves the intended fractional rate of spraying.

### CODE REVIEW

**Comment by dacap** [comment]
Hi @Fojar, I'll make some comments in the commits. (I haven't tested it yet.)

File: src/app/tools/point_shapes.h [hunk 1]
```diff
@@ -131,6 +131,7 @@ class FloodFillPointShape : public PointShape {
 
 class SprayPointShape : public PointShape {
   BrushPointShape m_subPointShape;
+  float point_remainder = 0;
```
Comment (dacap) [inline src/app/tools/point_shapes.h]: Rename to "m_pointRemainder" (similar to "m_subPointShape", we use "m_" prefix for member variables)

**Comment by dacap** [comment]
Could you squash both commits (and maybe the extra change) in just one commit?

**Comment by dacap** [comment]
As a side note, after I merge this PR, I'd love to change the random number generator with some of the new standard generators (e.g. Mersenne Twister). But that is something I'll add after this PR.

**Comment by dacap** [comment]
(Other note, don't worry about the Travis failed check, it was my fail with the freetype2 repository. It should be fixed now.)

**Comment by Fojar** [comment]
Sure, I will squash these commits. I'll have to make a new PR for that, correct?

**Comment by dacap** [comment]
>  I'll have to make a new PR for that, correct?

Not really, you can use `git push --force` to overwrite your old commits in your branch.

**Comment by Fojar** [comment]
Ah ok, I wasn't aware of that! Sorry, I already made a new PR, so I'll have to remember that for next time.

**Comment by dacap** [comment]
:+1: :smile:

**Pull request closed without merging**